pkg/util/spanlogger/spanlogger_test.go

```diff
@@ -19,3 +20,32 @@ func TestSpanLogger_Log(t *testing.T) {
 	require.Error(t, noSpan.Error(errors.New("err")))
 	require.NoError(t, noSpan.Error(nil))
 }
+
+func TestSpanLogger_CustomLogger(t *testing.T) {
+	var logged [][]interface{}
+	var logger funcLogger = func(keyvals ...interface{}) error {
+		logged = append(logged, keyvals)
+		return nil
+	}
+	span, ctx := NewWithLogger(context.Background(), logger, "test")
+	span.Log("msg", "original spanlogger")
+
+	span = FromContextWithFallback(ctx, log.NewNopLogger())
+	span.Log("msg", "restored spanlogger")
+
+	span = FromContextWithFallback(context.Background(), logger)
+	span.Log("msg", "fallback spanlogger")
+
+	expect := [][]interface{}{
+		{"method", "test", "msg", "original spanlogger"},
+		{"msg", "restored spanlogger"},
+		{"msg", "fallback spanlogger"},
```

Comment:
It's not clear to me if it's intentional that the method key is only attached to the original span logger and not the restored ones, but that's the way it behaved prior to my change.

Reply:
That is suspicious indeed, and perhaps a bug, but let's save that for a separate PR.

Reply:
On second thought, I think it makes sense even if it's not very intuitive: `FromContext`/`FromContextWithFallback` will be used from different methods than the one the span creates. I think it's probably correct that they're not annotated with the same method name since the logs are coming from different functions.